scripts/swap-beacon/src/index.ts

```diff
@@ -246,12 +249,25 @@ export class SwapBeacon {
     );
   }
 
+  heartbeat() {
+    if (this.pingTimeout !== undefined) clearTimeout(this.pingTimeout);
+
+    this.pingTimeout = setTimeout(() => {
+      console.error("Received no ping. Terminating connection.");
+      this.client.websocket.terminate();
+    }, this.PING_INTERVAL + 2000); // 2 seconds for latency
```

Comment:
if this terminates, does the session not just continue, with a broken WebSocket meaning no opportunities received? in that case, shouldn't we actually throw an error and exit the script, so that it can restart?

Reply:
nope, when we terminate the websocket the whole script exits. Seems a bit magical though